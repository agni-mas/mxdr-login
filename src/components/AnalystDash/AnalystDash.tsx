import HealthStatus from '../AdminDash/HealthStatus/HealthStatus';
import GridPageLayout from '../global/GridLayout/GridPageLayout';
import Table from '../global/table/Table';
import Grid from '../layout/Grid/Grid';
import GridItem from '../layout/GridItem/GridItem';
import AlertsAndReports from './AlertsAndReports/AlertsAndReports';
import IncidentTrends from './IncidentTrends/IncidentTrends';
import ServiceActivity from './ServiceActivity/ServiceActivity';
import ServiceSummary from './ServiceSummary/ServiceSummary'

const gridBoxData = [
  { num: 12, type: "Assigned to Me", isSelected: true },
  { num: 165, type: "Assigned to Org", isSelected: false },
  { num: 22, type: "Asset Alarms", isSelected: false },
  { num: 7, type: "Critical Alarms Today", isSelected: false },
  { num: 29, type: "Critical Incidents Today", isSelected: false },
];

const AnalystDash = () => (
  <section>
    <GridPageLayout {...gridBoxData} />
    <Grid>
      <GridItem title="Assigned To Me">
        <Table />
      </GridItem>
      <HealthStatus />
      <GridItem title='Incident Trends (last 7 days)'>
        <IncidentTrends />
      </GridItem>
      <GridItem title='New alerts and reports'>
        <AlertsAndReports />
      </GridItem>
      <GridItem title='Service Summary (Last 7 days) '>
        <ServiceSummary />
      </GridItem>
      <ServiceActivity />
    </Grid>
  </section>
);

export default AnalystDash;
