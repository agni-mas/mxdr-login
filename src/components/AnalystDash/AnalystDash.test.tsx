import { render } from "@testing-library/react";
import AnalystDash from "./AnalystDash";

it("renders <AnalystDash/>", () => {
  const { asFragment } = render(<AnalystDash />);
  expect(asFragment()).toMatchSnapshot();
});
