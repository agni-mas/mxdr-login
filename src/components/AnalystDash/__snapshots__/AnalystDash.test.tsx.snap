// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <AnalystDash/> 1`] = `
<DocumentFragment>
  <section>
    <div
      class="MuiBox-root css-i9gxme"
    >
      <div
        class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true css-1vd824g-MuiGrid-root"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-8nny99-MuiPaper-root"
          >
            <div
              style="padding-top: 13px; padding-left: 24px; padding-bottom: 13px; background-color: rgb(161, 0, 255);"
            >
              <div
                style="font-family: Graphik; font-style: normal; font-weight: 400; font-size: 32px; padding-bottom: 6px;"
              >
                12
              </div>
              <div
                style="font-family: Graphik; font-style: normal; font-weight: 400; font-size: 14px;"
              >
                Assigned to Me
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true css-1vd824g-MuiGrid-root"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-8nny99-MuiPaper-root"
          >
            <div
              style="padding-top: 13px; padding-left: 24px; padding-bottom: 13px;"
            >
              <div
                style="font-family: Graphik; font-style: normal; font-weight: 400; font-size: 32px; padding-bottom: 6px;"
              >
                165
              </div>
              <div
                style="font-family: Graphik; font-style: normal; font-weight: 400; font-size: 14px;"
              >
                Assigned to Org
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true css-1vd824g-MuiGrid-root"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-8nny99-MuiPaper-root"
          >
            <div
              style="padding-top: 13px; padding-left: 24px; padding-bottom: 13px;"
            >
              <div
                style="font-family: Graphik; font-style: normal; font-weight: 400; font-size: 32px; padding-bottom: 6px;"
              >
                22
              </div>
              <div
                style="font-family: Graphik; font-style: normal; font-weight: 400; font-size: 14px;"
              >
                Asset Alarms
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true css-1vd824g-MuiGrid-root"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-8nny99-MuiPaper-root"
          >
            <div
              style="padding-top: 13px; padding-left: 24px; padding-bottom: 13px;"
            >
              <div
                style="font-family: Graphik; font-style: normal; font-weight: 400; font-size: 32px; padding-bottom: 6px;"
              >
                7
              </div>
              <div
                style="font-family: Graphik; font-style: normal; font-weight: 400; font-size: 14px;"
              >
                Critical Alarms Today
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true css-1vd824g-MuiGrid-root"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-8nny99-MuiPaper-root"
          >
            <div
              style="padding-top: 13px; padding-left: 24px; padding-bottom: 13px;"
            >
              <div
                style="font-family: Graphik; font-style: normal; font-weight: 400; font-size: 32px; padding-bottom: 6px;"
              >
                29
              </div>
              <div
                style="font-family: Graphik; font-style: normal; font-weight: 400; font-size: 14px;"
              >
                Critical Incidents Today
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="grid"
    >
      <section
        class="item"
      >
        <h2
          class="grid-item-title"
        >
          Assigned To Me
        </h2>
        <div
          class="top-controls"
        >
          <div
            class="css-w4z10b-MuiStack-root"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSecondary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textSecondary MuiButton-sizeMedium MuiButton-textSizeMedium hover-state css-v87ghe-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              Incidents (0) 
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSecondary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textSecondary MuiButton-sizeMedium MuiButton-textSizeMedium hover-state css-v87ghe-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              Requests (0) 
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSecondary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textSecondary MuiButton-sizeMedium MuiButton-textSizeMedium hover-state css-v87ghe-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              Alarms (0) 
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSecondary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textSecondary MuiButton-sizeMedium MuiButton-textSizeMedium css-1d3f8j8-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Filter By
            <span
              class="MuiButton-endIcon MuiButton-iconSizeMedium css-9tj150-MuiButton-endIcon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="FilterAltIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M4.25 5.61C6.27 8.2 10 13 10 13v6c0 .55.45 1 1 1h2c.55 0 1-.45 1-1v-6s3.72-4.8 5.74-7.39c.51-.66.04-1.61-.79-1.61H5.04c-.83 0-1.3.95-.79 1.61z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="css-7yfvym-MuiStack-root"
        >
          <span
            class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-wave css-1kcnzpi-MuiSkeleton-root"
            style="height: 40px;"
          />
          <span
            class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-wave css-1kcnzpi-MuiSkeleton-root"
            style="height: 40px;"
          />
          <span
            class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-wave css-1kcnzpi-MuiSkeleton-root"
            style="height: 40px;"
          />
          <span
            class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-wave css-1kcnzpi-MuiSkeleton-root"
            style="height: 40px;"
          />
        </div>
      </section>
      <section
        class="item"
      >
        <h2
          class="grid-item-title"
        >
          Health Status
        </h2>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-disableElevation MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-disableElevation css-jqgntc-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Device Health
          <span
            class="MuiButton-endIcon MuiButton-iconSizeMedium css-9tj150-MuiButton-endIcon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="KeyboardArrowDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <span
          class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-wave css-1kcnzpi-MuiSkeleton-root"
          style="height: 200px;"
        />
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-wnpvqa-MuiButtonBase-root-MuiButton-root"
          style="float: right; border-radius: 28px;"
          tabindex="0"
          type="button"
        >
          View Details
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </section>
      <section
        class="item"
      >
        <h2
          class="grid-item-title"
        >
          Incident Trends (last 7 days)
        </h2>
        <div>
          <span
            class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-wave css-1kcnzpi-MuiSkeleton-root"
            style="height: 400px;"
          />
        </div>
      </section>
      <section
        class="item"
      >
        <h2
          class="grid-item-title"
        >
          New alerts and reports
        </h2>
        <div
          class="alerts"
        >
          <div
            class="MuiBox-root css-1gsv261"
          >
            <div
              class="MuiTabs-root css-1ujnqem-MuiTabs-root"
            >
              <div
                class="MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller"
                style="overflow: hidden; margin-bottom: 0px;"
              >
                <div
                  aria-label="basic tabs example"
                  class="MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer"
                  role="tablist"
                >
                  <button
                    aria-controls="simple-tabpanel-0"
                    aria-selected="true"
                    class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
                    color="secondary"
                    id="simple-tab-0"
                    role="tab"
                    tabindex="0"
                    type="button"
                  >
                    Alerts(0)
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                  <button
                    aria-controls="simple-tabpanel-1"
                    aria-selected="false"
                    class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
                    id="simple-tab-1"
                    role="tab"
                    tabindex="-1"
                    type="button"
                  >
                    Reports (0)
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </div>
                <span
                  class="MuiTabs-indicator css-1aquho2-MuiTabs-indicator"
                  style="left: 0px; width: 0px;"
                />
              </div>
            </div>
          </div>
          <span
            class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-wave css-1kcnzpi-MuiSkeleton-root"
            style="height: 400px;"
          />
          <div
            aria-labelledby="simple-tab-1"
            hidden=""
            id="simple-tabpanel-1"
            role="tabpanel"
          />
        </div>
      </section>
      <section
        class="item"
      >
        <h2
          class="grid-item-title"
        >
          Service Summary (Last 7 days) 
        </h2>
        <div
          class="bars"
        >
          <div>
            <h3
              class="bar-chart-title"
            >
              Incidents by urgency
            </h3>
            <div
              class="recharts-responsive-container"
              style="width: 100%; height: 250px;"
            />
          </div>
          <div>
            <h3
              class="bar-chart-title"
            >
              Requests by urgency
            </h3>
            <div
              class="recharts-responsive-container"
              style="width: 100%; height: 250px;"
            />
          </div>
        </div>
      </section>
      <section
        class="item"
      >
        <h2
          class="grid-item-title"
        >
          Service Activity (Last 7 days)
        </h2>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-disableElevation MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-disableElevation css-1oovqwt-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          My Org
          <span
            class="MuiButton-endIcon MuiButton-iconSizeMedium css-9tj150-MuiButton-endIcon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="KeyboardArrowDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <span
          class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-wave css-1kcnzpi-MuiSkeleton-root"
          style="height: 250px;"
        />
      </section>
    </div>
  </section>
</DocumentFragment>
`;
